server/util/trace_events/trace_events.go

```diff
@@ -0,0 +1,71 @@
+// The trace events package provides definitions and utilities for a subset of
+// Google's Trace Event Format that is relevant to BuildBuddy.
+//
+// The format is documented here:
+// https://docs.google.com/document/d/1CvAClvFfyA5R-PhYUmn5OOQtYMH4h6I0nSsKchNAySU
+package trace_events
```

Comment:
I originally tried using https://github.com/omaskery/teffy instead of writing this package, but ultimately didn't think it was worth it:
- The package nicely models the trace format but does so using struct embedding which IMO is too verbose and adds confusion.
- The `StreamingWriter` implementation doesn't append newlines between each JSON object which makes it non-browser-friendly. In the browser we rely on the newline, since the browser doesn't have a way to do a streaming JSON parse. (This is somewhat minor though and could be patched in if we really wanted it.)